### ISSUE TITLE
ignition hangs when downloading a file is required by container linux config

### ISSUE THREAD

**Issue by daveneeley** [coreos/bugs#2426]
# Issue Report #

### Container Linux Version ###
CoreOS stable (1688.5.3)

### Environment ###

What hardware/cloud provider/hypervisor is being used to run Container Linux?
The production vmware ova file was imported into a vmware vsphere 6.0.0 cluster.

### Expected Behavior ###
The coreos virtual machine comes up quickly after startup.

### Actual Behavior ###
The coreos virtual takes an indeterminate about of time, greater than 35 minutes in my current test run, to become available.

The vmware console shows the following messages repeatedly during this waiting period
```A start job is running for Ignition (files) (XXmin XXs / no limit)
files: createFilesystemsFiles: createFiles: op(7): GET https://github.com/docker/compose/releases/download/1.21.2/docker-compose-Linux-x86_64: attempt \#216
dial tcp: lookup github.com on [::1]:53: read udp [::1]:59108->[::1]:53: read connection refused
```

### Reproduction Steps ###

  1. Transpile a container linux config like the following (fake variables such as PRIVATE_IPV4 are all replaced prior to transpiling)
```networkd:
  units:
    - name: static.network
      contents:  |
        [Match]
        Name=en*

        [Network]
        Address={PRIVATE_IPV4}
        Gateway={PRIVATE_IPV4_GATEWAY}
        DNS={DNS1}
        DNS={DNS2}
        DHCP=no
        LinkLocalAddressing=no
        IPv6AcceptRA=no

storage:
  files:
    - path: /opt/bin/docker-compose
      filesystem: root
      mode: 0744
      contents:
        remote:
          url: https://github.com/docker/compose/releases/download/1.21.2/docker-compose-Linux-x86_64
          verification:
            hash:
              function: sha512
              sum: 5A9CCBE01CBF585732907F38FBACC21AECD2EDC9351BD1EE91DF750740A7FFF9D73BD489D708087D9D8892AD8F448A78A8EE401D240A51D0202ED0C08E30BF88
```
  2. Base64 encode the transpiled ignition.json file and add it to the appropriate guestinfo properties of the OVA before importing it
  3. Import the vm
  4. Switch to the vmware console and wait for the createFilesystemsFiles errors to begin

### Other Information ###

![image](https://user-images.githubusercontent.com/101746/39794013-58f7d18c-5305-11e8-8ae0-ab8604a5cc4e.png)

**Comment by lucab** [comment]
It looks like your environment may have some networking issues during initramfs. Please note that your `static.network` configuration only applies to the final system configuration, not to the initramfs.

If your environment doesn't rely on DHCP for node network configuration, are you passing the proper kernel cmdline as described in the [ignition documentation](https://coreos.com/ignition/docs/latest/network-configuration.html#using-static-ip-addresses-with-ignition)?

**Comment by daveneeley** [comment]
Thank you! I had not seen that bit of documentation on kernel cmdline for static ip. That is exactly how our environment functions. I see [adding kernel boot options](https://coreos.com/os/docs/latest/other-settings.html#adding-custom-kernel-boot-options) but neither this or [boot process](https://coreos.com/ignition/docs/latest/boot-process.html) describe how to inject these in an automated way so they are loaded in initramfs.

What is the recommended guidance for doing this?

**Comment by daveneeley** [comment]
The [vmware guestinfo](https://coreos.com/os/docs/latest/vmware-guestinfo.html) docs say these values I posted are processed by cloudinit, but isn't cloudinit deprecated? I am hoping there is a way to configure as much as possible in a pure container linux config transpiled to ignition.json.

**Comment by lucab** [comment]
Correct, those values are for the deprecated cloudinit, and they anyway take effects once the node is correctly provisioned (i.e. after initramfs stage).

In general we suggest to use DHCP to avoid ending up in this sort of chicken-egg problems with a node network configuration. In some environments (e.g. PXE) one can configure kernel command-line directly from the server, but I think this is not the case for ESXi. The other alternative is to manually adjust `grub.cfg` on the OEM partition *before* booting the node/disk for the first time, but it doesn't scale once you start having many nodes to boot.
